### PR TITLE
expression: fix extract bug when argument is a negative duration

### LINKED ISSUE #27236: `extract` result incompatible with mysql
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
select extract(hour_second from "-838:59:59.00");
```

### 2. What did you expect to see? (Required)
```sql
mysql> select  extract(hour_second from "-838:59:59.00");
+-------------------------------------------+
| extract(hour_second from "-838:59:59.00") |
+-------------------------------------------+
|                                  -8385959 |
+-------------------------------------------+
1 row in set (0.22 sec)
```
### 3. What did you see instead (Required)
```sql
MySQL > select  extract(hour_second from "-838:59:59.00");
+-------------------------------------------+
| extract(hour_second from "-838:59:59.00") |
+-------------------------------------------+
| 8385959                                   |
+-------------------------------------------+
1 row in set
Time: 0.051s
```
### 4. What is your TiDB version? (Required)
v5.0.3, v4.0.14,
master: da8bb5ea8c00bcbbb10afd0ae41a42fa
release-5.2.0: 4c68defe055a7040280

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #27236 

Problem Summary: `extract` didn't handle situation when its argument is a negative duration.

### What is changed and how it works?

How it Works:

### Check List 

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [x] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
expression: fix extract bug when argument is a negative duration
```

### CODE REVIEW

**Review by wshwsh12 — APPROVED** [review]
LGTM

File: expression/builtin_time.go [hunk 1]
```diff
@@ -2773,6 +2773,9 @@ func (b *builtinExtractDurationSig) evalInt(row chunk.Row) (int64, bool, error)
 		return 0, isNull, err
 	}
 	res, err := types.ExtractDurationNum(&dur, unit)
+	if dur.IsNeg() {
```
Comment (zanmato1984) [inline expression/builtin_time.go:2776]: I think maybe fixing the sign issue in function `types.ExtractDurationNum` is closer to the "root cause".

**Comment by zanmato1984** [comment]
Given that `types.ExtractDurationNum` has been updated, maybe this test https://github.com/pingcap/tidb/blob/79f0b495e5f70538de493fe868d921d234c2bde6/types/time_test.go#L1563 deserves some new cases?

**Comment by guo-shaoge** [comment]
/run-integration-common-test                                                                                                                                                                                                                                                  /run-common-test
/run-integration-copr-test

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Pull request merged**